gpm_api/tests/conftest.py

```diff
@@ -0,0 +1,322 @@
+import pytest
+import datetime
+from typing import Any, List, Dict, Tuple
+from gpm_api.io.products import get_info_dict, available_products
+import posixpath as pxp
+import ntpath as ntp
+import gpm_api.configs
+from unittest.mock import patch
+
+
+@pytest.fixture(scope="session", autouse=True)
+def mock_configuration():
+    """Patch the user configuration for entire session
+
+    Doing this will retrieve the configuration from pytest memory and not
+    alter the local configuration in ~/.config_gpm_api.yml
+    """
+
+    mocked_configuration = {
+        "gpm_username": "testuser",
+        "gpm_password": "testuser",
+        "gpm_base_dir": "data",
+    }
+
+    with patch.object(
+        gpm_api.configs,
+        "read_gpm_api_configs",
+        return_value=mocked_configuration,
+    ):
+        yield
+
+
+@pytest.fixture
+def product_types() -> List[str]:
+    """Return a list of all product types from the info dict"""
+    product_types = []
+    for product, props in get_info_dict().items():
+        product_types += props["product_types"]
+
+    product_types = list(set(product_types))  # Dedup list
+
+    return product_types
+
+
+@pytest.fixture
+def product_categories() -> List[str]:
+    """Return a list of product categories from the info dict"""
+
+    return list(set([props["product_category"] for props in get_info_dict().values()]))
+
+
+@pytest.fixture
+def product_levels() -> List[str]:
+    """Return a list of product levels from the info dict"""
+
+    # Available in gpm_api.io.checks.check_product_level()
+    return ["1A", "1B", "1C", "2A", "2B"]
+
+
+@pytest.fixture
+def versions() -> List[int]:
+    """Return a list of versions"""
+
+    # Available in gpm_api.io.checks.check_version()
+    return [4, 5, 6, 7]
+
+
+@pytest.fixture
+def products() -> List[str]:
+    """Return a list of all products regardless of type"""
+
+    return available_products()
+
+
+@pytest.fixture
+def username() -> str:
```

Comment:
If we will implement the access to the GES DISC HTTPS and AWS cloud buckets, we will need to add to the config files also the earthdata username and password. 
So maybe we will rename these arguments `pps_username` and `pps_password`.

Reply:
These fixtures became unnecessary with the mocked configuration, so I removed them and updated their corresponding test functions with the values that come from the configuration instead.